src/cost_export/function_app.py

```diff
@@ -408,77 +454,201 @@ def utilization_export_processor(msg: func.QueueMessage) -> None:
-                    failed_files.append({"file": blob.name, "error": str(e)})
+        # Process the specific blob from the message
+        try:
+            # Download blob content
+            blob_client = container_client.get_blob_client(blob_name)
+            blob_data = blob_client.download_blob().readall()
+            
+            # Transform S3 path
+            # Example blob_name: utilization-data/utilization-export/20250801-20250831/f2d6918d-67e2-4a1e-b5f6-9a69f1a87160/part_0_0001.csv.gz
+            # Final S3 path: {aws_target_file_path}/gds-recommendations-v1/billing_period=20250801/part_0_0001.csv.gz
+            path_parts = blob_name.split('/')
+            modified_parts = []
+            
+            for part in path_parts:
+                if part == "utilization-data" or part == "utilization-export":
+                    # Skip these parts entirely
                     continue
-        
-        # Summary logging
-        logging.info(f"Utilization processing complete. Successfully processed: {len(processed_files)} files")
-        if failed_files:
-            logging.warning(f"Failed to process: {len(failed_files)} files - {failed_files}")
-        
-        if not processed_files and not failed_files:
-            logging.info("No utilization CSV.GZ files found to process")
+                elif "-" in part and len(part) == 17 and part[:8].isdigit() and part[9:17].isdigit():
+                    # Transform date range (e.g., "20250801-20250831" -> "billing_period=20250801")
+                    billing_period = part.split("-")[0]
+                    modified_parts.append(f"billing_period={billing_period}")
+                elif len(part) == 36 and part.count('-') == 4 and all(c.isalnum() or c == '-' for c in part):
+                    # Skip GUID directories (format: 8-4-4-4-12 characters)
+                    logging.info(f"Skipping GUID directory: {part}")
+                    continue
+                else:
+                    modified_parts.append(part)
+            
+            modified_path = '/'.join(modified_parts)
+            s3_path = f"{s3_utilization_path.rstrip('/')}/{modified_path}"
+            
+            # Upload to S3
+            with s3.open_output_stream(s3_path) as f:
+                f.write(blob_data)
+            logging.info(f"Successfully uploaded {blob_name} to S3 at path: {s3_path}")
+            
+            # Delete source file after successful upload
+            blob_client.delete_blob()
+            logging.info(f"Successfully deleted source file: {blob_name}")
+            
+        except Exception as e:
+            logging.error(f"Failed to process {blob_name}: {str(e)}")
+            raise
             
     except Exception as e:
         logging.error(f"Error in utilization export processor: {str(e)}")
         raise
 
+@app.function_name(name="CarbonEmissionsBackfill")
+@app.route(route="carbon-backfill", auth_level=func.AuthLevel.FUNCTION)

```

Comment:
Worth moving the two functions into separate files if you can. If azure functions is anything like lambda you should be able to set the module path for the handler/function